lib/rules/require-sorted-object-keys.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Requires sorted object keys.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values:
+ *  - `true` (alias to `asc`)
+ *  - `"asc"`: requires sorting in ascending order
+ *  - `"desc"`: requires sorting in descending order
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireSortedObjectKeys": "asc"
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var x = {
+ *  x: 'foo',
+ *  y: 'bar'
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var x = {
+ *  y: 'foo',
+ *  x: 'bar'
+ * }
+ * ```
+ */
+
+function asc(a, b) {
+    if (/[_]/.test(a)) {
+        return -1;
+    }
+
+    if (a < b) {
+        return -1;
+    }
+
+    if (a > b) {
+        return 1;
+    }
+
+    return 0;
+}
+
+function desc(a, b) {
```

Comment:
The reason why a custom sort is needed (instead of `[].sort()` + `[].sort().reverse()`) is that underscores are considered words. This means that this object would be valid:

```js
var obj = {
  A: 'foo',
  _: 'qux'
  a: 'bar'
}
```

...since 65 (`A`) < 95 (`_`) < 97 (`a`). Stylistically, I don't think this is interesting, as symbols are typically sorted at the top or bottom. I believe this opinionated change is acceptable since in future anyone can create other sort functions if they don't find the current one fitting (e.g. `asc-native`). What are your thoughts?